app/services/hyrax/collection_types/create_service.rb

```diff
@@ -40,6 +42,20 @@ def self.create_collection_type(machine_id: DEFAULT_MACHINE_ID, title: DEFAULT_T
           c.assigns_workflow = opts[:assigns_workflow]
           c.assigns_visibility = opts[:assigns_visibility]
         end
+        add_participants(ct.id, opts[:participants])
+        ct
+      end
+
+      # @param collection_type_id [Integer]
+      # @param participants [Array]
+      def self.add_participants(collection_type_id, participants)
+        return unless collection_type_id && participants.count > 0
+        participants.each do |p|
+          agent_type = p[:agent_type]
+          agent_id = p[:agent_id]
+          access = p[:access]
+          Hyrax::CollectionTypeParticipant.create(hyrax_collection_type_id: collection_type_id, agent_type: agent_type, agent_id: agent_id, access: access)
```

Comment:
I believe this should be `.create!` instead of `.create`. If the underlying validation fails, I think that is a major thing to consider.

Reply:
Another consideration is that a particular combination should not be added twice.  A record is considered a duplicate when all four fields are identical...  collection type id, agent type, agent id, and access.

Reply:
There isn't any validation preventing adding the same combo twice. There doesn't seem to be in admin sets/permission templates either. Adding this validation would probably be a good idea.

Reply:
* Method was switched to `create!`
* As mentioned above, validation issues are covered by #1649.  This includes preventing dups.